language-design.md

```diff
@@ -28,28 +102,114 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
-
+It is much easier to spot a poorly-designed language than a well-designed
+language. While there are certain principles that can help you spot a
+well-designed language, time may be a factor in discovering a languages
+shortcomings.
+
+A well designed language should be approchable and "self-documenting." Just like
+APIs you should be able to understand the code without constantly looking at
+documentation once you know the pattern. This leads me to a second point:
+well-designed languages must have an overall pattern as the first paper
+mentioned [Steele, 1998]. Furthermore, as Steele explained, language extenstions
```

Comment:
Yep, Steele really emphasizes the important of putting in place a solid pattern in order to allow a language to grow. And this process of growth is ideally a communal one, in which the users themselves can implement extensions to the language yet maintaining the core structure of it. I thought another interesting argument that Steele mentioned was the importance of allowing the user to define new words that look and act like primitives of the language, which is something that Lisp particularly succeeded with.